### PR TITLE
Feature/justiceotuya/api connecting

### PR DESCRIPTION
**Pull request by justiceotuya**
Sorry This is much but i coded all day and am still coding

### CODE REVIEW

File: components/signup/components/SignupInputItemGenerator.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    REGISTER,
+} = STRINGS;
+
+/**
+* function that is used to also handle password validation, this compares the two password field;
+* @function
+* @param {function} actions actions attached to the input field for validation
+* @param {Object} items the values passed on to the input field
+* @param {function} decorator a Two-way binding for form
+* @param {string} label label for the input field
```
Comment (Way2nnadi) [inline components/signup/components/SignupInputItemGenerator.jsx]: nit: `label` not `label label`

File: components/signup/components/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        isFieldsValidating: PropTypes.func,
+        isSubmitting: PropTypes.func,
+        resetFields: PropTypes.func,
+        setFields: PropTypes.func,
+        setFieldsInitialValue: PropTypes.func,
+        setFieldsValue: PropTypes.func,
+        submit: PropTypes.func,
+        validateFields: PropTypes.func,
+        validateFieldsAndScroll: PropTypes.func,
+    }).isRequired,
```
Comment (Way2nnadi) [inline components/signup/components/SignUp.jsx]: I love the attention to documentation here, both i don't think you need to document `form` given that it is an `AntD` component and has it own set of documentation

Comment (Way2nnadi) [reply components/signup/components/SignUp.jsx]: `form: PropTypes.object.isRequired`

File: components/signup/components/SignupInputItemGenerator.jsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    case 'button':
+        Field = Button;
+        break;
+    case 'password':
+        Field = Input.Password;
+        break;
+    default:
+        Field = null;
+        break;
+    }
```
Comment (Way2nnadi) [inline components/signup/components/SignupInputItemGenerator.jsx:72]: can you wrap this switch statement in a method or function or something. It doesn't feel right being here

File: components/signup/sagas.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+function* handleUserRegistration({ payload }) {
+    const response = yield call(fetch, REGISTRATION_URL, {
+        body: JSON.stringify(payload),
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        method: 'POST',
+    });
+    const data = yield response.json();
+    if (response.ok) {
```
Comment (Way2nnadi) [inline components/signup/sagas.js:24]: you can't get the data is response is not ok

File: components/signup/sagas.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import {
+    takeEvery, call, put
+} from 'redux-saga/effects';
+import actionTypes from './actionTypes';
+import {
+    registrationRequestSuccess,
+    registrationRequestError
+} from './actions';
+
+const REGISTRATION_URL = '/api/users/register';
```
Comment (Way2nnadi) [inline components/signup/sagas.js]: move to `constants.js`

**Review by Way2nnadi — CHANGES_REQUESTED** [review]
Minor changes and potential error in saga

File: components/signup/components/SignUp.jsx (hunk 2 above)
Comment (justiceotuya) [reply components/signup/components/SignUp.jsx]: Eslit does not allow PropTypes.object

File: components/signup/components/SignupInputItemGenerator.jsx (hunk 3 above)
Comment (justiceotuya) [reply components/signup/components/SignupInputItemGenerator.jsx:72]: I really dont know hoe to fix this. i have wrapped my heads and nothing seems to come out. this switch statement was the result of a complex function and i dont really know what to do

Comment (Way2nnadi) [reply components/signup/components/SignupInputItemGenerator.jsx:72]: okay leave it for now

File: components/signup/components/SignUp.jsx (hunk 2 above)
Comment (Way2nnadi) [reply components/signup/components/SignUp.jsx]: that doesn't make any sense. `PropTypes.object` is valid (https://reactjs.org/docs/typechecking-with-proptypes.html)

**Review by Way2nnadi — APPROVED** [review]
LGTM

**Pull request merged**